pkg/prometheus/resource_selector.go

```diff
@@ -850,8 +850,8 @@ func (rs *ResourceSelector) validateKubernetesSDConfigs(ctx context.Context, sc
 			return fmt.Errorf("[%d]: %w", i, err)
 		}
 
-		if config.ProxyConfig != nil {
-			if err := validateProxyConfig(ctx, config.ProxyConfig, rs.store, sc.GetNamespace()); err != nil {
+		if !monitoringv1.IsProxyConfigEqualToZero(config.ProxyConfig) {
```

Comment:
(nit) I'm not against this abstraction, i think it might improve redability, but I would prefer if we move the negative operation to inside the function and maybe call this as `IsProxyConfigNotEqualToZero`